File: java/src/main/java/org/example/Cell.java
package org.example;

import java.util.HashSet;
import java.util.Set;

record Cell(
        int x,
        int y
) {

    boolean isNeighbour(Cell anotherCell) {
        return Math.abs(x - anotherCell.x) == 1 || Math.abs(y - anotherCell.y) == 1;
    }

    Set<Cell> calculatePossibleNeighbours() {
        Set<Cell> possibleNeighbours = new HashSet<>();
        for (int i = y-1; i <= y+1; i++) {
            for (int j = x-1; j <= x+1; j++) {
                if (i != y || j != x) {
                    possibleNeighbours.add(new Cell(j, i));
                }
            }
        }
        return possibleNeighbours;
    }
}

File: kotlin/src/main/kotlin/Cell.kt

internal data class Cell(val x: Int, val y: Int) {

    fun calculatePossibleNeighbours(): Set<Cell> {
        val possibleNeighbours: MutableSet<Cell> = HashSet()
        for (i in y - 1..y + 1) {
            for (j in x - 1..x + 1) {
                if (i != y || j != x) {
                    possibleNeighbours.add(Cell(j, i))
                }
            }
        }
        return possibleNeighbours
    }
}
